File: src/main/java/com/example/kyudo_app/domain/service/AssignBelongingGroupService.java
package com.example.kyudo_app.domain.service;

import com.example.kyudo_app.domain.model.BelongingGroup;
import com.example.kyudo_app.domain.model.User;
import com.example.kyudo_app.infrastructure.external.AssignBelongingGroupRepository;
import com.example.kyudo_app.infrastructure.external.UserRegisterRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class AssignBelongingGroupService {

    @Autowired
    private AssignBelongingGroupRepository assignBelongingGroupRepository;
    @Autowired
    private UserRegisterRepository userRegisterRepository;

    // group_idを使ってBelongingGroupを取得し、関連するエンティティにセット
    public void assignGroupToUserEntity(Long userId, Long groupId) {
        BelongingGroup belongingGroup = assignBelongingGroupRepository.findById(groupId)
                .orElseThrow(() -> new IllegalArgumentException("Invalid group ID: " + groupId));

        User user = userRegisterRepository.findById(userId)
                .orElseThrow(() -> new IllegalArgumentException("Invalid user ID: " + userId));

        user.setBelongingGroup(belongingGroup);
        userRegisterRepository.save(user);
    }
    }

File: src/main/kotlin/com/example/kyudo_app/domain/service/AssignBelongingGroupService.kt
package com.example.kyudo_app.domain.service

import com.example.kyudo_app.infrastructure.external.AssignBelongingGroupRepository
import com.example.kyudo_app.infrastructure.external.UserRegisterRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class AssignBelongingGroupService(
    private val assignBelongingGroupRepository: AssignBelongingGroupRepository,
    private val userRegisterRepository: UserRegisterRepository
) {
    // group_idを使ってBelongingGroupを取得し、関連するエンティティにセット
    fun assignGroupToUserEntity(userId: Long, groupId: Long?) {
        val belongingGroup = assignBelongingGroupRepository.findByIdOrNull(groupId)
            ?: throw IllegalArgumentException("指定されたグループが存在しません: $groupId")

        val user = userRegisterRepository.findByIdOrNull(userId)
            ?: throw IllegalArgumentException("Invalid user ID: $userId")

        user.setBelongingGroup(belongingGroup)
        userRegisterRepository.save(user)
    }
}
